src/tests/domain-test/user-unsubscription.test.ts

```diff
@@ -0,0 +1,70 @@
+import "mocha";
+import { expect } from "chai";
+import { UserDatasource } from "../../data/datasource/user.datasource";
+import { User } from "../../data/entity/user.entity";
+import { UserUnsubscription } from "../../domain/user-unsubscription.use-case";
+import { UserSubscription } from "../../domain/user-subscription.use-case";
+
+describe("Unsubscribe user", () => {
+  let userDatasource: UserDatasource;
+  let userSubscription: UserSubscription;
+  let userUnsubscription: UserUnsubscription;
+
+  before(() => {
+    userDatasource = new UserDatasource();
+    userUnsubscription = new UserUnsubscription();
+    userSubscription = new UserSubscription();
+  });
+
+  it("Should save user as active false when unsubscribing", async () => {
+    const user = new User();
+    user.firstName = "Rorschasch";
+    user.telegramId = 123445;
+    await userSubscription.subscribeUser(user);
+
+    await userUnsubscription.unsubscribeUser(user);
+    const saved = await userDatasource.findUserById(user.telegramId);
+
+    expect(saved!.isActive).to.be.false;
+  });
+
+  it("Should return 'User already subscribed", async () => {
+    const user = new User();
+    user.firstName = "Rorschasch";
+    user.telegramId = 123;
+    await userSubscription.subscribeUser(user);
+
+    try {
+      await userSubscription.subscribeUser(user);
+    } catch (error) {
+      expect(error.message).to.be.eq("User already subscribed.");
+      return;
+    }
+
+    expect.fail("Should have thrown an error");
```

Comment:
Acho que vc já tentou isso, mas o que acontece se fizer só 

```typescript
  await userSubscription.subscribeUser(user);
  expect.fail("User already subscribed.");
```

?

Reply:
O `expect.fail` força uma falha. Acho que é pra usar justamente numa condicional.